tests/e2e/README.md

```diff
@@ -0,0 +1,94 @@
+```
+CTR_TAG=not-latest make test-e2e
+```
+Note: If you use `latest` tag, and have not defined a registry, K8s will, by default, check the images on dockerhub, pulling and overriding the locally pushed images on the node.
+
+### Any other K8s deployment
+Have your `Kubeconf` pointing to your testing cluster of choice.
+The following code uses `latest` tag by default.
+```
+export CTR_REGISTRY=<myacr>.dockerhub.io # if needed, set CTR_REGISTRY_USER and CTR_REGISTRY_PASSWORD 
+make build-osm
+make docker-push
+go test ./tests/e2e -test.v -ginkgo.v -ginkgo.progress
+```
+
+## Flags
+### (TODO) Kubeconf selection
+Currently, test init will load a `Kubeconf` based on Defalut Kubeconf Loading rules. 
+If Kind is used, the kubeconf is temporarily replaced and Kind's kubeconf is used instead.
+
+### Container registry (required)
+A container registry where to load the images from (OSM, init container, etc.) is necessary. Can be set by flag or env value:
+```
+-ctrRegistry string
+		Container registry
+-ctrRegistrySecret string
+		Container registry secret
+-ctrRegistryUser string
+		Container registry username
+```
+If CR user and password are provided, the test framework will take care to add those as Docker secret credentials for the given container registry whenever appropriate (tenant namespaces for `init` containers, OSM intallation, etc).
+You can also set them through env:
+```
+export CTR_REGISTRY=<your_cr>.dockerhub.io
+export CTR_REGISTRY_USER=<uername>             # opt
+export CTR_REGISTRY_PASSWORD=<password>        # opt
+```
+
+### OSM Tag
+It will refer to the version of the OSM platform containers (OSM and init) for test to use:
+```
+-osmImageTag string
+		OSM image tag (default "latest")
+```
+Make sure you have compiled the images and pushed them on your registry first:
+```
+export CTR_REGISTRY=myacr.dockerhub.io
+export CTR_TAG=mytag               # Optional, 'latest' used by default
+make docker-push
+```
+
+### Use Kind for testing 
+Testing implements support for Kind. If `kindCluster` is enabled, a new Kind cluster will be provisioned and it will be automatically used for the test.
+```
+-kindCluster
+		Creates kind cluster
+-kindClusterName string
+		Name of the Kind cluster to be created (default "osm-e2e")
+-cleanupKindCluster
+		Cleanup kind cluster upon exit (default true)
+-cleanupKindClusterBetweenTests
+		Cleanup kind cluster between tests (default true)
+```
+
+### Test specific flags
+
+Worth mentioning `cleanupTest` is specially useful for debugging or leaving the test in a certain state at test-exit.
+When using Kind, you need to use `cleanupKindCluster` and `cleanupKindClusterBetweenTests` in conjunction, or else the cluster
+will anyway be destroyed.
+```
+-cleanupTest
+		Cleanup test resources when done (default true)
+-meshName string
+		OSM mesh name (default "osm-system")
+-waitForCleanup
+		Wait for effective deletion of resources (default true)
+```
+Plus, `go test` and `Ginkgo` specific flags, of course.
```

Comment:
I think it's mostly the registry, right? If instead of pushing images when using kind we used local registry (or registry url) there'd be virtually no difference between the two (and you would not suffer from the `latest` thing either?).

I feel we should keep your writting for overview. Reaching offline.